core/View.php

```diff
@@ -330,15 +330,17 @@ public function setContentType($contentType)
      * _Note: setting this allows you to make sure the View **cannot** be
      * embedded in iframes. Learn more [here](https://developer.mozilla.org/en-US/docs/HTTP/X-Frame-Options)._
      *
-     * @param string $option ('deny' or 'sameorigin')
+     * @param string $option ('deny' or 'sameorigin' or 'allow-from URI')
      */
     public function setXFrameOptions($option = 'deny')
     {
         if ($option === 'deny' || $option === 'sameorigin') {
             $this->xFrameOptions = $option;
+            return;
         }
-        if ($option == 'allow') {
-            $this->xFrameOptions = null;
+        if (stripos($option, 'allow-from') !== false) {
+            $this->xFrameOptions = $option;
```

Comment:
could you move this change to a separate PR? Ideally, a PR makes one fix only, and we use separate PRs for separate topics :+1: